### ISSUE TITLE
"Uncaught TypeError: next is not a function" in createMiddleware()

### ISSUE THREAD

**Issue by Craigson** [giantmachines/redux-websocket#44]
I'm getting the following: ``` Uncaught TypeError: next is not a function```

Which is referring to the ```return next(action)``` line in the createMiddleware() fn.

```
  /**
   * The primary Redux middleware function.
   * Each of the actions handled are user-dispatched.
   */
  return function (store) {
    return function (next) {
      return function (action) {
        switch (action.type) {
          // User request to connect
          case WEBSOCKET_CONNECT:
            close();
            initialize(store, action.payload);
            break;

          // User request to disconnect
          case WEBSOCKET_DISCONNECT:
            close();
            break;

          // User request to send a message
          case WEBSOCKET_SEND:
            if (websocket) {
              websocket.send(JSON.stringify(action.payload));
            } else {
              console.warn('WebSocket is closed, ignoring. Trigger a WEBSOCKET_CONNECT first.');
            }
            break;

          default:
            break;
        }
        return next(action);  // <==== Uncaught TypeError: next is not a function
      };
    };
  };
```

Creating my store as follows:

```
const websocketMiddleware = websocket({ prefix: 'TABLET' })

const middlewares = applyMiddleware(websocketMiddleware, thunk)

const storeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
const store = createStore(reducers, storeEnhancers(middlewares))
```

Any thoughts?

**Comment by brianmcallister** [comment]
At first glance, the only difference I see here is the integration with the Redux dev tools. Can you check to see that you're not getting the same exact error when _only_ using `thunk`? If not, we'll have to take a closer look at what's going on.

I didn't think it would be possible for a middleware in the stack to have control over what that `next` argument is (I thought it was managed fully by Redux and the only thing middleware had control over was the `action`, but I could be wrong about that).

**Comment by brianmcallister** [comment]
Oh and it looks like you're using the `1.0-dev` branch. Is that correct?

**Comment by Craigson** [comment]
Hi @brianmcallister, I can confirm that I **don't** get the same error when using only thunk.   And I was using package from npm, so I'm assuming 0.3.0?

**Comment by brianmcallister** [comment]
Got it. So, the currently published version is a little weird in that we're exporting the middleware function itself, and not a middleware creator function (like most other middleware). We're actively fixing that in the upcoming `1.0` release.

For the current `latest` version, _don't_ call the exported function, instead pass it directly to `applyMiddleware`.

Let me know how that works out.

**Comment by brianmcallister** [comment]
Hey @Craigson, I just published `1.0.0`, which will allow you to do this:

```js
const websocketMiddleware = websocket({ prefix: 'TABLET' });
```

Give it a shot and let me know how it works out!

**Comment by brianmcallister** [comment]
Closing this because it's been open for a while now, but @Craigson feel free to reopen if you're still experiencing difficulties.
